app/routes/questionnaire.py

```diff
@@ -544,14 +546,28 @@ def get_page_title_for_location(schema, current_location, context):
     return safe_content(page_title)
 
 
-def _render_page(block_type, context, current_location, previous_location_url, schema):
+def _render_page(
+    block_type,
+    context,
+    current_location,
+    previous_location_url,
+    schema,
+    questionnaire_store,
+):
     if request_wants_json():
         return jsonify(context)
 
     page_title = get_page_title_for_location(schema, current_location, context)
     session_data = get_session_store().session_data
     session_timeout = get_session_timeout_in_seconds(schema)
 
+    are_hub_required_sections_complete = all(
```

Comment:
We should move this logic out of this method - it can be added to questionnaire store updater which is available on the block

Reply:
Nothing should be added to questionnaire store updater as no state is being updated.

Reply:
We've left it where it was.

Reply:
> Nothing should be added to questionnaire store updater as no state is being updated.

Fair point. Do you think this is the best place for these methods though?